sphinx.conf

```diff
@@ -15,20 +22,65 @@ source repos {
     sql_attr_bool = is_archived
     sql_attr_bool = is_mirror
     sql_attr_bool = is_empty
-    sql_query = SELECT id, hosting_service_id, name, username, description, extract(epoch from created_at) as created_at, extract(epoch from updated_at) as updated_at, extract(epoch from pushed_at) as pushed_at, is_fork, is_archived, is_empty, is_mirror FROM new_repositories
 }
 
-index repos {
-    source                  = repos
-    path                    = /var/lib/sphinx/data/repos_index
+source repos_p1 : repos {
+    sql_query = $BASE_QUERY where id %% 5 = 0
+}
+source repos_p2 : repos {
+    sql_query = $BASE_QUERY where id %% 5 = 1
+}
+source repos_p3 : repos {
+    sql_query = $BASE_QUERY where id %% 5 = 2
+}
+source repos_p4 : repos {
+    sql_query = $BASE_QUERY where id %% 5 = 3
+}
+source repos_p5 : repos {
+    sql_query = $BASE_QUERY where id %% 5 = 4
+}
+
+index repos_p1 {
+    source                  = repos_p1
+    path                    = /var/lib/sphinx/data/repos_p1
     morphology              = stem_en
 }
+index repos_p2 {
+    source                  = repos_p2
+    path                    = /var/lib/sphinx/data/repos_p2
+    morphology              = stem_en
+}
+index repos_p3 {
+    source                  = repos_p3
+    path                    = /var/lib/sphinx/data/repos_p3
+    morphology              = stem_en
+}
+index repos_p4 {
+    source                  = repos_p4
+    path                    = /var/lib/sphinx/data/repos_p4
+    morphology              = stem_en
+}
+index repos_p5 {
+    source                  = repos_p5
+    path                    = /var/lib/sphinx/data/repos_p5
+    morphology              = stem_en
+}
+
+index repos {
+    type = distributed
+    local = repos_p1
+    local = repos_p2
+    local = repos_p3
+    local = repos_p4
+    local = repos_p5
+}
```

Comment:
...and we create a "distributed" index, consisting of our repo partitions

Reply:
thats a... funny looking key/value thing, how does it keep them apart =P